### PR TITLE
start writing calendar code

### PR DESCRIPTION
**Pull request by hannahsalameh**
Is there a way to do what I am doing without converting it from an arraylist to an array?

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -28,45 +27,46 @@
...
-    int latestEndTime = eventsArray[0].getWhen().end();
-    for(int i = 0; i < eventsArray.length; ++i){
-        if(i == 0){
-            int firstEventStart = eventsArray[i].getWhen().start();
-            if((firstEventStart - TimeRange.START_OF_DAY) >= request.getDuration()){
-                meetingTimes.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, firstEventStart, false));
-            }
+    //sort by begin time
+    //eventsArrayList could be more specific 
+    //initialize > populate, not initialize > initilaize > populate
```
Comment (ofk-goog) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Looks like these are notes on changes you intended to make. Some changes you've made, some you haven't.

**Review by ofk-goog — CHANGES_REQUESTED** [review]
Looks like you accidentally added a bunch of unrelated changes to this PR.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 1 above)
Comment (hannahsalameh) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: sorry about that! I'm still working on it

Comment (hannahsalameh) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: ok should be good now. I'm not sure what the unrelated push was

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -19,39 +19,37 @@
...
     //check first event start time against beginning of day
-    int firstEventStart = eventsArrayList.get(0).getWhen().start();
+    int firstEventStart = conflictingEventTimes.get(0).start();
     if((firstEventStart - TimeRange.START_OF_DAY) >= request.getDuration()){
         meetingTimes.add(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, firstEventStart, false));
     }
 
-    int latestEndTime = eventsArrayList.get(0).getWhen().end();
+    int latestEndTime = conflictingEventTimes.get(0).end();
     //checks all middle events to see if there is enough time between them
```
Comment (ofk-goog) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Ooh, I just realized this is a bug. The event with the latest start time may not have the latest end time.

Consider also adding a test case for this.

**Review by ofk-goog — CHANGES_REQUESTED** [review]
Looks great. But there's one bug.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 2 above)
Comment (hannahsalameh) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: I'm comparing latest end times, not start times. That's why I track the end time w another variable. I can add a test case for it still, though.

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java [hunk 3]
```diff
@@ -223,6 +297,56 @@ public void justEnoughRoom() {
...
   }
 
+  @Test
+  public void justEnoughRoomOptional() {
+    // Have one person, but make it so that there is just enough room at one point in the day to
+    // have the meeting.
+    //
+    // Events  : |--A--|     |----A----|
+    // Day     : |---------------------|
+    // Options :       |-----|
```
Comment (ofk-goog) [inline walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java]: This description isn't quite right. It doesn't mention PERSON_C.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 4]
```diff
@@ -24,14 +24,27 @@
     if(request.getDuration() > TimeRange.WHOLE_DAY.duration()){
         return meetingTimes;
     }
-    
-    ArrayList<TimeRange> conflictingEventTimes = new ArrayList<TimeRange>();
+
+    boolean countOptional = true;
```
Comment (ofk-goog) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: This doesn't need to be a variable. Just use `true` and `false` inline

**Review by ofk-goog — CHANGES_REQUESTED** [review]
Good. Just a couple minor changes.

**Review by ofk-goog — APPROVED** [review]

**Pull request merged**